src/js/angular/core/services/workbench-context.service.js

```diff
@@ -0,0 +1,180 @@
+            callback(getSelectedRepositoryId());
+        }
+        return EventEmitterService.subscribeSync(WorkbenchEventName.SELECTED_REPOSITORY_ID_UPDATED, (payload) => callback(payload));
+    };
+
+    /**
+     * Updates the "autocompleteEnabled" flag and emits the 'autocompleteEnabledUpdated' event to notify listeners that the autocomplete status is changed.
+     *
+     * @param {boolean} autocompleteEnabled
+     */
+    const setAutocompleteEnabled = (autocompleteEnabled) => {
+        if (_autocompleteEnabled !== autocompleteEnabled) {
+            _autocompleteEnabled = autocompleteEnabled;
+            EventEmitterService.emitSync(WorkbenchEventName.AUTOCOMPLETE_ENABLED_UPDATED, getAutocompleteEnabled());
+        }
+    };
+
+    /**
+     *
+     * @return {boolean} the status of autocomplete.
+     */
+    const getAutocompleteEnabled = () => {
+        return _autocompleteEnabled;
+    };
+
+    /**
+     * Subscribes to the 'autocompleteEnabledUpdated' event.
+     * @param {function} callback - The callback to be called when the event is fired.
+     *
+     * @return {function} unsubscribe function.
+     */
+    const onAutocompleteEnabledUpdated = (callback) => {
+        if (angular.isFunction(callback)) {
+            callback(getAutocompleteEnabled());
+        }
+        return EventEmitterService.subscribeSync(WorkbenchEventName.AUTOCOMPLETE_ENABLED_UPDATED, (payload) => callback(payload));
+    };
+
+    /**
+     * Updates the selected repository namespaces and emits the 'selectedRepositoryNamespacesUpdated' event to notify listeners that the selected repository namespaces are changed.
+     *
+     * @param {NamespacesListModel} selectedRepositoryNamespaces
+     */
+    const setSelectedRepositoryNamespaces = (selectedRepositoryNamespaces) => {
+        if (!isEqual(_selectedRepositoryNamespaces, selectedRepositoryNamespaces)) {
+            _selectedRepositoryNamespaces = cloneDeep(selectedRepositoryNamespaces);
+            EventEmitterService.emitSync(WorkbenchEventName.SELECTED_REPOSITORY_NAMESPACES_UPDATED, getSelectedRepositoryNamespaces());
+        }
+    };
+
+    /**
+     *
+     * @return {NamespacesListModel} the selected repository namespaces.
+     */
+    const getSelectedRepositoryNamespaces = () => {
+        return cloneDeep(_selectedRepositoryNamespaces);
+    };
+
+    /**
+     * Subscribes to the 'selectedRepositoryNamespacesUpdated' event.
+     * @param {function} callback - The callback to be called when the event is fired.
+     *
+     * @return {function} unsubscribe function.
+     */
+    const onSelectedRepositoryNamespacesUpdated = (callback) => {
+        if (angular.isFunction(callback)) {
+            callback(getSelectedRepositoryNamespaces());
+        }
+        return EventEmitterService.subscribeSync(WorkbenchEventName.SELECTED_REPOSITORY_NAMESPACES_UPDATED, (payload) => callback(payload));
+    };
+
+    const resetContext = () => {
+        init();
```

Comment:
Remove this and do the reset here and don't forget the _selectedRepositoryId which you omitted for some reason in the init function